x-pack/solutions/security/plugins/elastic_assistant/server/lib/prompt/get_prompt.ts

```diff
@@ -0,0 +1,187 @@
+  promptGroupId: string;
+  promptIds: string[];
+}
+
+type PromptArray = Array<{ promptId: string; prompt: string }>;
+/**
+ * Get prompts by feature
+ * provide either model + provider or connector to avoid additional calls to get connector
+ * @param actionsClient - actions client
+ * @param connector - connector, provide if available. No need to provide model and provider in this case
+ * @param connectorId - connector id
+ * @param model - model. No need to provide if connector provided
+ * @param promptGroupId - feature id, should be common across promptIds
+ * @param promptIds - prompt ids with shared promptGroupId
+ * @param provider  - provider. No need to provide if connector provided
+ * @param savedObjectsClient - saved objects client
+ */
+export const getPromptsByGroupId = async ({
+  actionsClient,
+  connector,
+  connectorId,
+  model: providedModel,
+  promptGroupId,
+  promptIds,
+  provider: providedProvider,
+  savedObjectsClient,
+}: GetPromptsByGroupIdArgs): Promise<PromptArray> => {
+  const { provider, model } = await resolveProviderAndModel(
+    providedProvider,
+    providedModel,
+    connectorId,
+    actionsClient,
+    connector
+  );
+
+  const prompts = await savedObjectsClient.find<Prompt>({
+    type: promptSavedObjectType,
+    searchFields: ['promptId'],
+    search: `${promptGroupId}-*`,
+  });
+  const promptsOnly = prompts?.saved_objects.map((p) => p.attributes) || [];
+
+  return promptIds.map((promptId) => ({
+    promptId,
+    prompt:
+      findPromptEntry(
+        promptsOnly.filter((p) => p.promptId === promptId) || [],
+        promptId,
+        provider,
+        model
+      ) || '',
+  }));
+};
+
+/**
+ * Get prompt by promptId
+ * provide either model + provider or connector to avoid additional calls to get connector
+ * @param actionsClient - actions client
+ * @param connector - connector, provide if available. No need to provide model and provider in this case
+ * @param connectorId - connector id
+ * @param model - model. No need to provide if connector provided
+ * @param promptId - prompt id
+ * @param provider  - provider. No need to provide if connector provided
+ * @param savedObjectsClient - saved objects client
+ */
+export const getPrompt = async ({
+  actionsClient,
+  connector,
+  connectorId,
+  model: providedModel,
+  promptId,
+  provider: providedProvider,
+  savedObjectsClient,
+}: GetPromptArgs): Promise<string> => {
+  const { provider, model } = await resolveProviderAndModel(
+    providedProvider,
+    providedModel,
+    connectorId,
+    actionsClient,
+    connector
+  );
+
+  const prompts = await savedObjectsClient.find<Prompt>({
+    type: promptSavedObjectType,
+    searchFields: ['promptId'],
+    search: promptId,
+    fields: ['provider', 'model', 'prompt'],
+  });
+
+  return (
+    findPromptEntry(
+      prompts?.saved_objects.map((p) => p.attributes) || [],
+      promptId,
+      provider,
+      model
+    ) || ''
```

Comment:
I'm wondering if callers of `getPrompt` would expect it to resolve with an empty string or reject if a default prompt was not available. WDYT?

Reply:
Not sure, didn't think we'd want to block the request over it